scipy/ndimage/morphology.py

```diff
@@ -535,6 +535,17 @@ def binary_opening(input, structure=None, iterations=1, output=None,
         By default, a new array is created.
     origin : int or tuple of ints, optional
         Placement of the filter, by default 0.
+    mask : array_like, optional
```

Comment:
Since all these are copies of the text in the erosion/dilation functions, it may be a good idea to move the docstring text to `_ni_docstrings.py` and use the `@docfiller` decorator from there.

Reply:
Found an issue with the `@_ni_doctrings.docfiller` decorator. It gets confused by any `%` characters in the docstring. In my case, I have the following url in the documentation of the binary_opening function: `[1] http:://en.wikipedia.org/wiki/Opening%28morphology%29`.

When doing a `import scipy.ndimage` I get the following error:
```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/user/scipy/scipy/ndimage/__init__.py", line 164, in <module>
    from .measurements import *
  File "/home/user/scipy/scipy/ndimage/measurements.py", line 38, in <module>
    from . import morphology
  File "/home/user/scipy/scipy/ndimage/morphology.py", line 512, in <module>
    origin=0, mask=None, border_value=0, brute_force=False):
  File "/home/user/scipy/scipy/misc/doccer.py", line 218, in decorate
    f.__doc__ = docformat(f.__doc__, docdict)
  File "/home/user/scipy/scipy/misc/doccer.py", line 68, in docformat
    return docstring % indented
TypeError: not enough arguments for format string
```
If I remove the `%` chars with `[1] http:://en.wikipedia.org/wiki/Opening^28morphology^29`, I get no errors.

So for now, I'll leave the documentation as it is, copy-pasted from other functions, and go ahead with the testing changes and the binary_closing function. What do you want me to do about the decorator?

Reply:
We should probably switch it over to the `.format()` method, but that is clearly something to do in another PR, not this.